### DISCUSSION TITLE
does mediamtx support growing files?

### DISCUSSION THREAD

**Discussion by coolapso** [bluenviron/mediamtx#3499] [Questions]
### Question

Hi, not sure if this is already written somehwere or not, couldn't really find anything relevant. 

When recording it looks like mediamtx keeps the segment in memory before flushing it into disk, this means that if I want a segment 1h segment of fmp4 it will keep the whole hour in memory which becomes a problem when using it with SBC that have limited amount of memory to work with. Is it possible to tell mediamtx to keep writing to the file similar to what ffmpeg does when you just set input > output or like OBS or other softwares?  or am i doing something wrong? 

basically what I would like to accomplish is ...

Set recording > record to file till the file contains 1h of video > close file > start writing to new one

Hope I am being clear of what I mean.

**Comment by aler9** [comment]
Hello, 

> When recording it looks like mediamtx keeps the segment in memory before flushing it into disk, this means that if I want a segment 1h segment of fmp4 it will keep the whole hour in memory

This is not true, segments are flushed to disk as soon as `recordPartDuration` is reached. The default `recordPartDuration` is 1 second, therefore segments are flushed every second.

> is it possible to tell mediamtx to keep writing to the file similar to what ffmpeg does when you just set input > output or like OBS or other softwares? or am i doing something wrong?

there's no need to support growing files since you can generate on the fly recording segments of any duration by using the [playback server](https://github.com/bluenviron/mediamtx?tab=readme-ov-file#playback-recorded-streams).

Enable the playback server and then request a segment. The server will take care of concatenating or truncating existing segments in order to reach the desired length, and then it will return the result:

```
http://localhost:9996/list?path=mypath
http://localhost:9996/get?path=mypath&start=2024-01-14T16%3A33%3A17%2B00%3A00&duration=200.5
```

[a comment by an automated account is not shown]
